src/dynamic_honey_badger/dynamic_honey_badger.rs

```diff
@@ -315,14 +588,35 @@ where
         let (key_gen, part) = SyncKeyGen::new(our_id, sk, pub_keys, threshold)?;
         self.key_gen_state = Some(KeyGenState::new(key_gen, change.clone()));
         if let Some(part) = part {
-            let step_on_send = self.send_transaction(KeyGenMessage::Part(part))?;
-            step.extend(step_on_send);
+            step.extend(self.send_transaction(KeyGenMessage::Part(part))?);
         }
         Ok(step)
     }
 
+    /// Moves queued messages from Honey Badger to Dynamic Honey Badger.
+    fn move_queued_hb_messages(&mut self) {
+        for (key, hb_msgs) in self.honey_badger.outgoing_queue() {
+            let (id, hb_epoch) = key;
+            for hb_msg in hb_msgs.drain(..) {
+                let msg = from_hb_message(hb_msg, self.start_epoch);
+                let e = self
+                    .outgoing_queue_hb
+                    .entry((id.clone(), DynamicEpoch::new(self.start_epoch, *hb_epoch)));
+                match e {
+                    btree_map::Entry::Occupied(e) => {
+                        e.into_mut().push(msg);
+                    }
+                    btree_map::Entry::Vacant(e) => {
+                        e.insert(vec![msg]);
+                    }
+                }
+            }
+        }
+    }
+
     /// Starts a new `HoneyBadger` instance and resets the vote counter.
     fn restart_honey_badger(&mut self, epoch: u64) -> Result<Step<C, N>> {
+        self.move_queued_hb_messages();
```

Comment:
We can actually discard all Honey Badger messages whose HB epoch is greater than `epoch - self.start_epoch`, I think: All nodes will restart their HB instance at this point?